.npmrc

```diff
@@ -0,0 +1,6 @@
+# .npmrc
+# Configuration for npm and pnpm
+
+# Uses the exact version instead of any within-patch-range version of an
+# installed package
+save-exact=true
```

Comment:
What's your thinking here? (We're already doing `npm ci` for our tests, which installs exact dependencies and sub-dependencies from package-lock.json.)

Reply:
oh, actually, we're doing it for the top-level, but not the external tests. But still, we could change that if e wanted to.

Reply:
`npm ci` is only for the tests, however, when someone new who wants to contribute runs `npm install`, npm will install any version within the patch range instead of exactly that version, and the person will commit the version bump. To prevent this, the `save-exact` removes the `^` from `package.json`.

A couple of tests fail on node 12 on some version of jest and typescript when I accidently bumped deps like this a week ago, hence I put this in to prevent it from happening in the future.

Reply:
Eh, I use `npm ci` locally, but maybe I'm just weird 🙃 

Does `save-exact` prevent changes to `package-lock.json` when running `npm install`?

Fair enough, though. And, if current versions of jest or whatever else we depend on no longer work in node.js 12, then perhaps we should drop it from CI.

Reply:
> Does save-exact prevent changes to package-lock.json when running npm install?

I think it does, take a look at some of the answers [here](https://stackoverflow.com/questions/45022048/why-does-npm-install-rewrite-package-lock-json)

> Fair enough, though. And, if current versions of jest or whatever else we depend on no longer work in node.js 12, then perhaps we should drop it from CI.

I haven't spend much time on it so I'm not exactly sure what is causing the trouble, will look into it and let you know